### PR TITLE
CB-10628 Fix emulate android --target

### PR DESCRIPTION
**Pull request by daserge**
[Jira issue](https://issues.apache.org/jira/browse/CB-10628)

### CODE REVIEW

**Comment by nikhilkh** [comment]
How can we test this?

File: bin/templates/cordova/lib/run.js [hunk 1]
```diff
@@ -41,8 +41,9 @@ var path  = require('path'),
 
     var self = this;
 
-    var install_target = runOptions.device ? '--device' :
-        runOptions.emulator ? '--emulator' :
+    var install_target = !runOptions.target ?
```
Comment (vladimir-kotikov) [inline bin/templates/cordova/lib/run.js]: IMO this statement looks very complex - there is 3 ternary operators in one expression. Could you please try to simplify this

**Comment by daserge** [comment]
@vladimir-kotikov, addressed, thanks.
@nikhilkh, I've tried to test private `getInstallTarget` with `rewire` but faced an issue caused by the fact that the test-time directory structure differs from the run-time one (so that `check-reqs` fails to load when rewire requires run -> emulator -> check_reqs):

```javascript
var rewire = require("rewire");
var run = rewire("../../bin/templates/cordova/lib/run");
var getInstallTarget = run.__get__("getInstallTarget");

describe("run", function () {
  describe("getInstallTarget", function() {
    var targetOpts = { target: "emu" };
    var deviceOpts = { device: true };
    var emulatorOpts = { emulator: true };
    var emptyOpts = {};

    it("should select correct target based on the run opts", function() {
        expect(getInstallTarget(targetOpts)).toBe("emu");
        expect(getInstallTarget(deviceOpts)).toBe("--device");
        expect(getInstallTarget(emulatorOpts)).toBe("--emulator");
        expect(getInstallTarget(emptyOpts)).toBeUndefined();
    });
  });
});
```

**Comment by daserge** [comment]
I will take a look at this once more tomorrow.

**Comment by daserge** [comment]
Worked on this with @vladimir-kotikov - there is a way to solve this using `require.cache` but it looks like an overkill for this case. 
Decided to require `check_reqs` at runtime instead.

**Comment by daserge** [comment]
@nikhilkh, please take a look.

**Comment by nikhilkh** [comment]
LGTM

**Pull request merged**